src/types/currency.ts

```diff
@@ -20,7 +20,20 @@ function isoToBytes(iso: string): Buffer {
  * Tests if ISO is a valid iso code
  */
 function isIsoCode(iso: string): boolean {
-  return iso.length === 3;
+  return ISO_REGEX.test(iso);
+}
+
+function isoCodeFromHex(code: Buffer): string | undefined {
+  const iso = code.toString();
```

Comment:
Does `Buffer.toString()` remove all leading and trailing zeros?

Reply:
This is only looking at the 3 bytes in the middle, not the whole string